### PR TITLE
*: consistent replica read

### PR DESCRIPTION
**Pull request by nolouch**
## What have you changed? (mandatory)
Ref: https://github.com/tikv/tikv/issues/1280
Let TiKV can do read on the follower, use `read_index` to ensure consistent reading.  RFC:   https://github.com/tikv/rfcs/pull/14

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- New feature

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -573,6 +594,10 @@ impl Peer {
         self.raft_group.raft.state == StateRole::Leader
     }
 
+    pub fn is_follower(&self) -> bool {
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: we need to support learner here.

**Comment by siddontang** [comment]
please rename the title to consistent replica read.

**Comment by overvenus** [comment]
PTAL @nolouch  , should we close this PR?

**Comment by overvenus** [comment]
This PR is being closed because it is inactive.

**Pull request closed without merging**